solidity-checks/src/main/java/org/sonarsource/solidity/checks/CheckUtils.java

```diff
@@ -66,14 +65,12 @@ public static boolean treeMatches(ParseTree tree, Class context) {
   }
 
   public static Optional<ParseTree> checkForElseStatement(ParserRuleContext ctxNode) {
-    if (!ctxNode.children.isEmpty() && ctxNode.children.size() >= 6) {
-      // the 6th child is where else exists even for else-if case
-      Token token = (Token) ctxNode.children.get(5).getPayload();
-      if (token.getType() == 41) {
-        ParseTree child6 = ctxNode.children.get(6);
+    if (!ctxNode.children.isEmpty()) {
+      if ((((IfStatementContext) ctxNode).Else()) != null) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=63&issues=AWSxeJdL6dA5b-W2oUml&open=AWSxeJdL6dA5b-W2oUml)

[ISSUE_KEY]: AWSxeJdL6dA5b-W2oUml